Packages/com.llealloo.audiolink/Runtime/Scripts/YtdlpPlayer.cs

```diff
@@ -242,29 +272,94 @@ public override void OnInspectorGUI()
+                        if (EditorGUI.EndChangeCheck())
+                            _ytdlpPlayer.SetPlaybackTime(playbackTime);
+
+                        // Timestamp input
+                        EditorGUI.BeginChangeCheck();
+                        double time = hasVideoPlayer ? _ytdlpPlayer.VideoPlayer.time : 0;
+                        string currentTimestamp = " " + _ytdlpPlayer.FormattedTimestamp(time);
+                        string seekTimestamp = EditorGUILayout.DelayedTextField(currentTimestamp, GUILayout.MaxWidth(8 * currentTimestamp.Length));
+                        if (EditorGUI.EndChangeCheck() && videoLength > 0)
+                        {
+                            TimeSpan inputTimestamp;
+                            // Add an extra 00: to force TimeSpan to interpret 12:34 as 00:12:34 for proper mm:ss input
+                            if (TimeSpan.TryParse($"00:{seekTimestamp}", out inputTimestamp))
+                            {
+                                playbackTime = (float)(inputTimestamp.TotalSeconds / videoLength);
+                                _ytdlpPlayer.SetPlaybackTime(playbackTime);
+                            }
+                        }
+                    }
+
+                    // Media Controls
+                    using (new EditorGUILayout.HorizontalScope())
+                    {
+                        bool isPlaying = hasVideoPlayer ? _ytdlpPlayer.VideoPlayer.isPlaying : false;
+                        bool isPaused = hasVideoPlayer ? _ytdlpPlayer.VideoPlayer.isPaused : false;
+                        bool isStopped = !isPlaying && !isPaused;
+
+                        bool play = GUILayout.Toggle(isPlaying, new GUIContent(" Play", EditorGUIUtility.IconContent("d_PlayButton On").image), "Button") != isPlaying;
+                        bool pause = GUILayout.Toggle(isPaused, new GUIContent(" Pause", EditorGUIUtility.IconContent("d_PauseButton On").image), "Button") != isPaused;
+                        bool stop = GUILayout.Toggle(isStopped, new GUIContent(" Stop", EditorGUIUtility.IconContent("d_Record Off").image), "Button") != isStopped;
+
+                        if (hasVideoPlayer)
+                        {
+                            if (play)
+                                _ytdlpPlayer.VideoPlayer.Play();
+                            else if (pause)
+                                _ytdlpPlayer.VideoPlayer.Pause();
+                            else if (stop)
+                                _ytdlpPlayer.VideoPlayer.Stop();
+                        }
+                    }
+                }
+
+                float volume;
+                using (new EditorGUI.DisabledScope(!_ytdlpPlayer.GetAudioSourceVolume(out volume)))
+                {
+                    EditorGUI.BeginChangeCheck();
+                    volume = EditorGUILayout.Slider(new GUIContent("  AudioSource Gain", EditorGUIUtility.IconContent("d_Profiler.Audio").image), volume, 0.0f, 1.0f);
```

Comment:
Labelling this "AudioSource Gain" in the UI led me originally to believe this would affect AudioLink internal gain only. A more canonical name like "Volume" might be better for UX.

Reply:
this makes sense to me as it's also called Volume in the AudioSource component